google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -4586,4 +4587,217 @@ public void testPreserveAsciiControlCharacters()
     assertEquals("\u0000", row.get(0).getStringValue());
     assertTrue(bigquery.delete(tableId));
   }
+
+  @Test
+  public void testReferenceFileSchemaUriForAvro() {
+    try {
+      String destinationTableName = "test_reference_file_schema_avro";
+      TableId tableId = TableId.of(DATASET, destinationTableName);
+      Schema expectedSchema =
+          Schema.of(
+              Field.newBuilder("username", StandardSQLTypeName.STRING)
+                  .setMode(Mode.NULLABLE)
+                  .build(),
+              Field.newBuilder("tweet", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build(),
+              Field.newBuilder("timestamp", StandardSQLTypeName.STRING)
+                  .setMode(Mode.NULLABLE)
+                  .build(),
+              Field.newBuilder("likes", StandardSQLTypeName.INT64).setMode(Mode.NULLABLE).build());
+
+      // By default, the table should have c-twitter schema because it is lexicographically last.
+      // a-twitter schema (username, tweet, timestamp, likes)
+      // b-twitter schema (username, tweet, timestamp)
+      // c-twitter schema (username, tweet)
+      List<String> SOURCE_URIS =
+          ImmutableList.of(
+              "gs://"
+                  + CLOUD_SAMPLES_DATA
+                  + "/bigquery/federated-formats-reference-file-schema/a-twitter.avro",
+              "gs://"
+                  + CLOUD_SAMPLES_DATA
+                  + "/bigquery/federated-formats-reference-file-schema/b-twitter.avro",
+              "gs://"
+                  + CLOUD_SAMPLES_DATA
+                  + "/bigquery/federated-formats-reference-file-schema/c-twitter.avro");
+
+      // Because referenceFileSchemaUri is set as a-twitter, the table will have a-twitter schema
+      String referenceFileSchema =
+          "gs://"
+              + CLOUD_SAMPLES_DATA
+              + "/bigquery/federated-formats-reference-file-schema/a-twitter.avro";
+
+      LoadJobConfiguration loadJobConfiguration =
+          LoadJobConfiguration.newBuilder(tableId, SOURCE_URIS, FormatOptions.avro())
+              .setReferenceFileSchemaUri(referenceFileSchema)
+              .build();
+
+      Job job = bigquery.create(JobInfo.of(loadJobConfiguration));
+      // Blocks until this load table job completes its execution, either failing or succeeding.
+      job = job.waitFor();
+      if (job.isDone()) {
```

Comment:
this logic seems odd (here and in the other variants).  A job can be done with or without an error attached.  you can just assert the job completed without errors?

Also, as this doesn't appear to be a sample it's not clear what emitting to system.out is getting you.

Reply:
fixed